docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6328,409 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+data types which are not covered by P4Data. This incorporates all the data types
+used as primitive data types like bytes, strings.
+Some of them have default values. These default values are only applicable
+when in a Union field and not for Match fields. Match fields do not have
+default values because if a match field is not present, then it is considered
+to be a wildcard match. But if a union member is not present, then the default
+value is used. Numeric Default values have been kept as `bytes` so that
+lengths > 64 can be easily accommodated.
+
+  * bitstring : Bytes type used for simple unsigned integers or bytes. Default
+    value applicable. If not present, it is 0.
+  * bool : Boolean type values. Default value applicable. If not present, it
+    is `false`.
+  * generic_struct : Used to define a structure that can contain
+    one or more of the other members as named fields. One structure can contain
+    different members of different types in GenericData.
+  * generic_list : Used to define a list of values with the same type. It can
+    also be a list of generic_structs or generic_list.
+    Members are ordered and duplicates are allowed.
+  * generic_unordered_set : Used to define a set of same types. It can also be
+    a set of generic_structs. Members are unordered and duplicates are not
+    allowed.
+  * string : Used to define control plane strings. The max string length is
+    defined by the size. Default value applicable. If not present, it is empty
+    string: "".
+  * enum : String representation with which safe enums are realized. Default
+    value applicable. If not present, it is empty string: "".
+  * enum_val : bytes value with which unsafe/serializable enums are defined.
+    Default value applicable. If not present, it is 0.
+
+## `GenericTableEntry` { #sec-generic-table-entry}
+
+GenericTableEntry can be used to program non-PSA externs or non-P4
+target-specific "fixed" features to their implementation in a generic way. This
+provides an alternative to `Extern` in a structured way and within p4runtime
+guidelines and semantics.  The idea has been borrowed from TDI [Table Driven
+Interface](https://github.com/p4lang/tdi), that every state can be representated
+as a table or multiple tables. A table here is any data structure with one or
+more entries and each entry can be represented as a set of key fields and data
+fields.  Actions in P4 tables behave like unions and that has been borrowed here
+in `GenericTable`.  This can be potentially used targets to map to either TDI
+based targets, or even non-TDI based targets. If using TDI, then potentially,
+for every new feature to be added, only TDI target support and remote client
+code will be required to make use of that function.  All other middle layers
+should theoretically remain unchanged.
+
+P4Runtime supports inserting, modifying, deleting and reading `GenericTable`
+entries with the `GenericTa bleEntry` entity, which has the following fields:
```

Comment:
typo: incorrect space in middle of `GenericTableEntry`

Reply:
suggested replacement: "entity" -> "message" ?  It is not a big deal either way, but doesn't Protobuf use "message" pretty consistently for things like `GenericTableEntry`?